geode-core/src/test/java/org/apache/geode/test/dunit/rules/ClusterStartupRule.java

```diff
@@ -304,19 +320,55 @@ public VM getVM(int index) {
     return getHost(0).getVM(index);
   }
 
-  public void stopVM(int index) {
-    stopVM(index, true);
+  /**
+   * gracefully stop the member inside this vm
+   *
+   * @param index vm index
+   */
+  public void stopMember(int index) {
+    stopMember(index, true);
   }
 
-  public void stopVM(int index, boolean cleanWorkingDir) {
+  /**
+   * gracefully stop the member inside this vm
+   */
+  public void stopMember(int index, boolean cleanWorkingDir) {
     VMProvider member = occupiedVMs.get(index);
 
     if (member == null)
       return;
 
-    member.stopVM(cleanWorkingDir);
+    member.stopMember(cleanWorkingDir);
   }
 
+  /**
+   * this disconnects the distributed system of the member. The member will automatically try to
+   * reconnect after 5 seconds.
+   *
+   * will throw a ClassCastException if this method is called on a client VM.
+   */
+  public void disconnectMember(int index) {
+    MemberVM member = getMember(index);
+    if (member == null)
+      return;
+    member.disConnectMember();
+  }
+
+  /**
+   * this crashes the VM hosting the member/client. It removes the VM from the occupied VM list
+   * so that we can ignore it at cleanup.
+   */
+  public void crashVM(int index) {
+    VMProvider member = occupiedVMs.remove(index);
+    member.invokeAsync(() -> {
```

Comment:
This will leave orphaned JVMs around if there are any non-daemon threads like a CacheServer AcceptorImpl thread or a Locator TcpServer thread.  I think you need to do more work here to ensure that any cache contained in the JVM is crashed.  DistributedTestUtils.crashDistributedSystem(), for example.

Reply:
even after I call System.exist(), the non-daemon threads are still up?

Reply:
Non-daemon threads will not exit if you invoke System.exit() and they will keep the JVM from shutting down.

Reply:
According to the java doc of Thread. When the exit method is called, the JVM stops executing all the threads and terminates.

When a Java Virtual Machine starts up, there is usually a single non-daemon thread (which typically calls the method named main of some designated class). The Java Virtual Machine continues to execute threads until either of the following occurs:

* The exit method of class Runtime has been called and the security manager has permitted the exit operation to take place.
* All threads that are not daemon threads have died, either by returning from the call to the run method or by throwing an exception that propagates beyond the run method.

Reply:
In our cache servers only the AcceptorImpl thread keeps the JVM from exiting.  Check it out.
https://www.j2eeonline.com/java-tm-fundamentals-II/module4/setDaemon-threads.jsp

Reply:
Also, in our distributed unit tests there is a non-daemon "main" thread.  AcceptorImpl, "main" and any other dangling non-daemon threads created by tests will keep the JVM from exiting.  That's what the second bullet point in the Thread javadoc is saying.

Reply:
There is a better mechanism to crash a JVM - ProcessManager.bounce().  ProcessManager has the actual process objects for the JVMs and can kill them.

Reply:
I made more changes to this method. We call bounce immediately after we made sure the vm is crashed, which essentially is calling ProcessManager.bounce().